### ISSUE TITLE
Intended behavior of `TorchModuleWrapper`?

### ISSUE THREAD

**Issue by AlexanderLavelle** [keras-team/keras#20000]
I am trying to implement the `keras.layers.TorchModuleWrapper` as a way to align use Torch modules in my Keras network with target JAX backend.

tl;dr - with the wrapper, I get the following error
```
'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'dim'

Arguments received by TorchModuleWrapper.call():
  • args=('tf.Tensor(shape=(96, 12), dtype=int64)',)
  • kwargs=<class 'inspect._empty'>
```

Dim is an attempted call on a Torch tensor.

If I convert my batch to `torch.tensor` and run it through the unwrapped layer, I get the anticipated result. When I pass the same `torch.tensor` through the wrapped layer and it is converted to tf tensor, dim is no longer a method of `tf.Variable` or...in this case `EagerTensor`

Perhaps I am not correctly interpreting the usage of this wrapper?

**Comment by AlexanderLavelle** [comment]
After a little more modifications, this outputs the values but it's a poor solution for so many reasons

- why use one library when you can use jax, torch, and tf
- can only get .numpy() in eager mode, and in graph mode cannot use `tf.get_static_value` on the tensor
- likely that autograd is broken because of the torch tensor detach

```python
Class TorchModuleWrapper(Layer):
    ...
    @staticmethod
    def tensor_converter(item):
        import tensorflow as tf
        import torch
        output = item
        if isinstance(item, tf.Tensor):
            output = torch.tensor(item.numpy())
        return output

    def call(self, *args, **kwargs):
        import tensorflow as tf
        from jax.tree_util import tree_map
        args = list(map(self.tensor_converter, args))
        kwargs = tree_map(self.tensor_converter, kwargs)
        return tf.constant(self.module(*args, **kwargs).detach().numpy())
    ...
```

**Comment by grasskin** [comment]
Hi @AlexanderLavelle, `TorchModuleWrapper` is meant for running vanilla Pytorch with Keras set to Torch backend. It does not currently support Jax as a backend since the intent was to allow adding non Keras torch components to a Keras model on the Torch backend, in the future this might be supported. Thank you for raising this!
